src/main/java/com/stack/project/Impl/stackImpl.java

```diff
@@ -0,0 +1,62 @@
+package com.stack.project.Impl;
+
+import java.util.Vector;
+
+import com.stack.project.stack;
+
+public class stackImpl <E> implements stack<E>  {
+
+    private Vector<E> vec;
```

Comment:
change it to "T arr[]" and not vector, we want to implement something here and not use a already functioning and defined data structure. 

add capacity + top variables to help you with all the functions implementation. you need to think now how to do it. also please make sure you allocate on the heap the "E arr[]"

Reply:
fixed